packages/server/src/koaJsonRpc/index.ts

```diff
@@ -110,9 +113,10 @@ export default class KoaJsonRpc {
         return;
       }
 
+      this.requestId = ctx.state.reqId;
       const methodName = body.method;
       const methodTotalLimit = this.registryTotal[methodName];
-      if (this.rateLimit.shouldRateLimit(ctx.ip, methodName, methodTotalLimit)) {
+      if (this.rateLimit.shouldRateLimit(ctx.ip, methodName, methodTotalLimit, this.requestId)) {
         ctx.body = jsonResp(body.id, new IPRateLimitExceeded(methodName), undefined);
         ctx.status = 409;
```

Comment:
```suggestion
        ctx.status = 429;
```
nit: should we change, as 429 error code is for too many requests

Reply:
Agreed, however the relay client used for testing is built using ethers.js.  The ethers.js provider consumes the 429 error code and uses it to implement throttling.  We would need to implement our own ethers.js provider, much like infura or alchemy have done.

Reply:
Gotcha. That seems like a work for a new issue, maybe.

Reply:
Does this mean we're still throwing a 409 instead of the more appropriate 429 that we agreed on?

If so yeah agree with Georgi looks like future work.
Please open a ticket around updating to 429.
We'll probably have to use a separate test flow not ethersjs.
Maybe just a raw HTTP request client that confirms it